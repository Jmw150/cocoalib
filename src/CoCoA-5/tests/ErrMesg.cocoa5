R ::= QQ[x,y,z];
K := NewFractionField(R);

try
  println "toric: check exponents June2018 -- Bug #1190";
  N := 9;
  M := mat([[ random(-99,99) | j in 1..2*N] | i in 1..N]);
  H := HilbertBasisKer(M);
uponerror e do
  println GetErrMesg(e); println;
endtry

try
  println "subst now check indets 1Mar2017";
  P1 ::= QQ[x,y];
  P2 ::= QQ[x,y];
  f := RingElem(P1, "x+2*y");
  subst(f, RingElem(P2, "x"), 1); 
uponerror e do
  println GetErrMesg(e); println;
endtry

try
  println "gcd in field now throws Error 24May2012";
  gcd(zero(K), RingElem(K,"x")) = 1;
uponerror e do
  println GetErrMesg(e); println;
endtry

try
  println "gcd in field now throws Error 24May2012";
  gcd(zero(K), zero(K)) = 0;
uponerror e do
  println GetErrMesg(e); println;
endtry


try
  println "Hilbert fn: check for positive grading Oct2019 -- Bug #1211";
    R := NewPolyRing(QQ, "x,y,z", MakeTermOrdMat(RowMat([2,3,0])), 1);
    HilbertSeries(R/ideal(RingElem(R, "z")));
uponerror e do
  println GetErrMesg(e); println;
endtry

