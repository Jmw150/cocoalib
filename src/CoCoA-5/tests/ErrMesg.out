toric: check exponents June2018 -- Bug #1190
exponents too big in binomial
--> [CoCoALib] MatKerToBListAndIndices

subst now check indets 1Mar2017
x is not an element of domain

gcd in field now throws Error 24May2012
Ring is not a true GCD domain (e.g. it is a field)
--> [CoCoALib] gcd(x,y)

gcd in field now throws Error 24May2012
Ring is not a true GCD domain (e.g. it is a field)
--> [CoCoALib] gcd(x,y)

Hilbert fn: check for positive grading Oct2019 -- Bug #1211
Grading must be positive
--> [CoCoALib] HilbertSeriesQuot

